cli/types/cypress.d.ts

```diff
@@ -1899,18 +1899,27 @@ declare namespace Cypress {
      * @see https://on.cypress.io/screenshot
      * @example
      *    cy.screenshot()
-     *    cy.get(".post").screenshot()
      */
     screenshot(options?: Partial<Loggable & Timeoutable & ScreenshotOptions>): Chainable<null>
     /**
      * Take a screenshot of the application under test and the Cypress Command Log and save under given filename.
      *
      * @see https://on.cypress.io/screenshot
      * @example
-     *    cy.get(".post").screenshot("post-element")
+     *    cy.screenshot("post-element")
      */
     screenshot(fileName: string, options?: Partial<Loggable & Timeoutable & ScreenshotOptions>): Chainable<null>
 
+    /**
+     * Take a screenshot of the application under test and the Cypress Command Log and save under given filename.
+     * When this method is chained it will also pass the context along to the screenshot.
+     *
+     * @see https://on.cypress.io/screenshot
+     * @example
+     *    cy.get(".post").screenshot("post-element")
+     */
+    screenshot(context: string, fileName?: string, options?: Partial<Loggable & Timeoutable & ScreenshotOptions>): Chainable<null>,
```

Comment:
I'm assuming the `context` parameter is meant to represent the `subject` from the previously-chained command? I think that should look more like this (note the `Chainable<Subject>` instead of `Chainable<null>`) based on examples elsewhere in this file

```ts
screenshot(fileName?: string, options?: Partial<Loggable & Timeoutable & ScreenshotOptions>): Chainable<Subject>,
```

Reply:
Hey sorry! Yea I'm having a hard time getting the yarn install to work something about python versions and some windows tools. That's why I marked it as a draft because I haven't actually been able to run and build this locally yet. 

I also saw that [this check](https://github.com/cypress-io/cypress/pull/27130/checks?check_run_id=14518877915) was failing and I wasn't sure what steps to take to resolve that. Any help on either of those things would be appreciated!

I'll update to fix the `Chainable<Subject>` and the changelog. I actually don't know why I called that variable context haha. I'll change it to subject.

I'll add those test cases as well, granted that I figure out this issue with running locally. If you have some suggestions for me to try i'd be happy to. Otherwise this might have to wait till later this week when I'm a little more available. 

Thanks for taking a look @mike-plummer

Reply:
Ready for another pass I think!